commit 392cb8075fe5ae2f20b9fdf0d85e4510cc799cc1
Author: Cole Robinson <crobinso@redhat.com>
Date:   Thu Oct 13 16:59:15 2011 -0400

    connection: Don't have tick edit object lists in place
    
    Since 'tick' isn't really thread safe, and we sometimes call a manual tick
    to refresh resources (say immediately after creating a guest), we
    can't depend on lists staying consistent.

diff --git a/src/virtManager/connection.py b/src/virtManager/connection.py
index fdd5ec8..9fb8335 100644
--- a/src/virtManager/connection.py
+++ b/src/virtManager/connection.py
@@ -1278,7 +1278,7 @@ class vmmConnection(vmmGObject):
         return (stop, start, origlist, new, current)
 
     def _update_nets(self):
-        orig = self.nets
+        orig = self.nets.copy()
         name = "network"
         active_list = self.vmm.listNetworks
         inactive_list = self.vmm.listDefinedNetworks
@@ -1291,7 +1291,7 @@ class vmmConnection(vmmGObject):
                                  lookup_func, build_class)
 
     def _update_pools(self):
-        orig = self.pools
+        orig = self.pools.copy()
         name = "pool"
         active_list = self.vmm.listStoragePools
         inactive_list = self.vmm.listDefinedStoragePools
@@ -1304,7 +1304,7 @@ class vmmConnection(vmmGObject):
                                  lookup_func, build_class)
 
     def _update_interfaces(self):
-        orig = self.interfaces
+        orig = self.interfaces.copy()
         name = "interface"
         active_list = self.vmm.listInterfaces
         inactive_list = self.vmm.listDefinedInterfaces
@@ -1318,7 +1318,7 @@ class vmmConnection(vmmGObject):
 
 
     def _update_nodedevs(self):
-        orig = self.nodedevs
+        orig = self.nodedevs.copy()
         name = "nodedev"
         active_list = lambda: self.vmm.listDevices(None, 0)
         inactive_list = lambda: []
@@ -1340,7 +1340,7 @@ class vmmConnection(vmmGObject):
         oldActiveIDs = {}
         oldInactiveNames = {}
 
-        origlist = self.vms
+        origlist = self.vms.copy()
         current = {}
         new = []
 
